include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            radSplashType,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            "unitSI",
-                            &factorDirection);
-
-                        /* position */
-                        std::vector<float_X> positionMesh(simDim, 0.0); /* there is no offset - zero */
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            splashFloatXType,
-                            (meshesPathName + dataLabelsDetectorDirection(detectorDim)).c_str(),
-                            "position",
-                            1u,
-                            splash::Dimensions(simDim, 0, 0),
-                            &(*positionMesh.begin()));
+                            // select data
+                            for(int copyIndex = 0; copyIndex < N_tmpBuffer; ++copyIndex)
+                            {
+                                tmpBuffer[copyIndex]
+                                    = ((picongpu::float_64*)
+                                           values.data())[ampIndex + Amplitude::numComponents * copyIndex];
+                            }
+                            // write actual data
+                            mesh_amp[dir].storeChunk(::openPMD::shareRaw(tmpBuffer), offset_amp, extent_amp);
+                            openPMDdataFile.flush();
+                        }
                     }
+                    // end: write amplitude data
+
+
+                    // start: write observer
+                    // prepare n meshes
+                    ::openPMD::Mesh mesh_n = openPMDdataFileIteration.meshes[dataLabelsDetectorDirection(-1)];
+                    // write mesh attributes
+                    mesh_n.setGeometry(::openPMD::Mesh::Geometry::cartesian); // set be default
+                    // mesh_n.setGeometryParameters( geometryParameters );
+                    mesh_n.setDataOrder(::openPMD::Mesh::DataOrder::C);
+                    mesh_n.setGridSpacing(std::vector<double>{1.0, 1.0, 1.0});
+                    mesh_n.setGridGlobalOffset(std::vector<double>{0.0, 0.0, 0.0});
+                    mesh_n.setGridUnitSI(1.0);
+                    mesh_n.setAxisLabels(std::vector<std::string>{"detector direction index", "None", "None"});
+                    mesh_n.setUnitDimension(std::map<::openPMD::UnitDimension, double>{});
 
+                    {
+                        std::vector<float_64> tmpBuffer(parameters::N_observer);
 
-                    /* save detector frequencies */
-                    splash::Dimensions bufferSizeOmega(1, radiation_frequencies::N_omega, 1);
+                        // reshape abstract MeshRecordComponent
+                        ::openPMD::Datatype datatype_n = ::openPMD::determineDatatype(::openPMD::shareRaw(tmpBuffer));
```

Comment:
```suggestion
                        ::openPMD::Datatype datatype_n = ::openPMD::determineDatatype<float_64>();
```
or do you want to make that depend on `tmpBuffer`? Don't know if `determineDatatype<typename decltype(tmpBuffer)::value_type>()` is more legible, but it's at least clearer on what you're trying to do.

Reply:
No, it should stay with `float_64` so your (first) suggestion is perfect - thanks.